layers/utils.py

```diff
@@ -0,0 +1,72 @@
+import json
+import os
+import subprocess
+
+
+def run(args):
+    print(">", " ".join(args))
+    subprocess.run(args, check=True)
+
+
+def ensureEmptyTempDirectoryExists(directoryName):
+    if os.path.isdir(directoryName):
+        run(["rm", "-r", directoryName])
+    os.makedirs(directoryName, exist_ok=True)
+
+
+def convertPbfToGeoJson(pbfPath, geojsonPath, geometryType):
+    run(
+        [
+            "osmium",
+            "export",
+            pbfPath,
+            f"--geometry-type={geometryType}",
+            "-o",
+            geojsonPath,
+        ]
+    )
+
+
+# Note the layer name is based on the input filename. This always generates
+# numeric feature IDs.
+def convertGeoJsonToPmtiles(geojsonPath, pmtilesPath):
+    run(["tippecanoe", geojsonPath, "--generate-ids", "-o", pmtilesPath])
+
+
+# Adds numeric IDs to every feature, trims coordinate precision, and uses the
+# callback to transform each feature's properties. Overwrites the file.
+#
+# The callback takes (input properties, output properties) and doesn't return
```

Comment:
Really tempted to switch to Node+TS for this script ^^;  It takes more words to describe what a type signature does easily